streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -111,6 +111,12 @@ public StateStoreSupplier build() {
 
                                     @Override
                                     public PersistentKeyValueFactory<K, V> windowed(final long windowSize, final long retentionPeriod, final int numSegments, final boolean retainDuplicates) {
+                                        if (numSegments < 1) {
```

Comment:
Since we are already checking this in `RocksDBWindowStoreSupplier`, do we still need the same checking here?